src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -70,13 +88,48 @@ private void ActivateProperties(object controller, Type containerType, Dictionar
             {
                 var propertyHelper = propertyHelpers.First(helper =>
                     string.Equals(helper.Name, property.Key, StringComparison.Ordinal));
-                if (propertyHelper.Property == null || !propertyHelper.Property.CanWrite)
+                if (propertyHelper.Property == null)
+                {
+                    // Nothing to do if property is unknown to PropertyHelper.
+                    continue;
+                }
+
+                if (propertyHelper.Property.CanWrite)
+                {
+                    // Handle settable property.
+                    propertyHelper.SetValue(controller, property.Value);
+                    continue;
+                }
+
+                var propertyType = propertyHelper.Property.PropertyType;
+                if (propertyType.IsArray)
+                {
+                    // Do not attempt to copy values into an array because an array's length is immutable. This choice
+                    // is also consistent with MutableObjectModelBinder's handling of a read-only array property.
+                    continue;
+                }
+
+                var source = property.Value;
+                var target = propertyHelper.GetValue(controller);
+                if (source == null || target == null)
+                {
+                    // Nothing to do when source or target is null.
+                    continue;
+                }
+
+                // Determine T if this is an ICollection<T> property.
+                var elementTypeArray = propertyType
```

Comment:
nit: `elementTypeArray` gives me the impression that this is the element type **of** an array. Suggestion: `collectionTypeArguments`